Frends.Kafka.Produce/Frends.Kafka.Produce/Produce.cs

```diff
@@ -1,212 +1,411 @@
+                return await ProduceBasic(input.Topic, input.Partition, input.Key, input.Message, producerConfig, cancellationToken);
+        }
+        catch (Exception)
+        {
+            throw;
+        }
+    }
+
+    private static async Task<Result> ProduceBasic(string topic, int partition, string msgKey, string msg, ProducerConfig producerConfig, CancellationToken cancellationToken)
+    {
+        using var producer = new ProducerBuilder<string, string>(producerConfig).Build();
+
+        TopicPartition topicPartition = null;
+        if (partition >= 0)
+            topicPartition = new TopicPartition(topic, new Partition(partition));
+
+        Message<string, string> message = new()
+        {
+            Key = msgKey ?? null,
+            Value = msg
+        };
+
+        if (topicPartition != null)
+            return new Result(true, await producer.ProduceAsync(topicPartition, message, cancellationToken));
+        else
+            return new Result(true, await producer.ProduceAsync(topic, message, cancellationToken));
+    }
+
+    private static async Task<Result> ProduceAvro(string topic, string msgKey, int partition, SchemaRegistry schemaRegistry, ProducerConfig producerConfig, CancellationToken cancellationToken)
+    {
+        var schemaRegistryConfig = new SchemaRegistryConfig()
+        {
+            Url = AssignIfNotNullOrEmpty(schemaRegistry.SchemaRegistryUrl, string.Empty),
+            BasicAuthCredentialsSource = GetBasicAuthCredentialsSource(schemaRegistry.BasicAuthCredentialsSource),
+            EnableSslCertificateVerification = schemaRegistry.EnableSslCertificateVerification,
+            BasicAuthUserInfo = AssignIfNotNullOrEmpty(schemaRegistry.BasicAuthUserInfo, string.Empty),
+            RequestTimeoutMs = schemaRegistry.RequestTimeoutMs,
+            MaxCachedSchemas = schemaRegistry.MaxCachedSchemas,
+        };
+
+        if (!string.IsNullOrEmpty(schemaRegistry.SslCaLocation))
+        {
+            schemaRegistryConfig.SslCaLocation = AssignIfNotNullOrEmpty(schemaRegistry.SslCaLocation, string.Empty);
+            schemaRegistryConfig.SslKeystorePassword = AssignIfNotNullOrEmpty(schemaRegistry.SslKeystorePassword, string.Empty);
+            schemaRegistryConfig.SslKeystoreLocation = AssignIfNotNullOrEmpty(schemaRegistry.SslKeystoreLocation, string.Empty);
+        }
 
-            Message<string, string> message = new()
+        TopicPartition topicPartition = null;
+        if (partition >= 0)
+            topicPartition = new TopicPartition(topic, new Partition(partition));
+
+        using var cachedSchemaRegistryClient = new CachedSchemaRegistryClient(schemaRegistryConfig);
+        using var producer = new ProducerBuilder<string, GenericRecord>(producerConfig)
+            .SetValueSerializer(new AvroSerializer<GenericRecord>(cachedSchemaRegistryClient))
+            .Build();
+
+        var avroSchema = (RecordSchema)Avro.Schema.Parse(schemaRegistry.SchemaJson);

```

Comment:
Ok, so just to clarify (and tell me if I am mistaking something) - it should be possible to just get the schema from the registry itself, without providing your own schema as a JSON for the task. Is this an option that we can do here as well?